lms/models/users.py

```diff
@@ -0,0 +1,28 @@
+import re
+
+from flask_babel import gettext as _  # type: ignore
+
+from lms.lmsdb.models import User
+from lms.models.errors import (
+    ForbiddenPermission, UnauthorizedError, UnhashedPasswordError,
+)
+
+
+def retrieve_salt(user: User) -> str:
+    HASHED_PASSWORD = re.compile(
+        r'^pbkdf2.+?\$(?P<salt>.+?)\$(?P<password>.+)',
+    )
```

Comment:
The compilation should be in the global scope (the whole idea is to improve the performance)